pkg/controller/volume/persistentvolume/scheduler_binder.go

```diff
@@ -277,12 +276,30 @@ func (b *volumeBinder) BindPodVolumes(assumedPod *v1.Pod) error {
 		return err
 	}
 
-	return wait.Poll(time.Second, b.bindTimeout, func() (bool, error) {
+	if err := wait.Poll(time.Second, b.bindTimeout, func() (bool, error) {
 		// Get cached values every time in case the pod gets deleted
 		bindings = b.podBindingCache.GetBindings(assumedPod, assumedPod.Spec.NodeName)
 		claimsToProvision = b.podBindingCache.GetProvisionedPVCs(assumedPod, assumedPod.Spec.NodeName)
 		return b.checkBindings(assumedPod, bindings, claimsToProvision)
-	})
+	}); err != nil {
+		return err
+	}
+
+	node, err := b.ctrl.kubeClient.CoreV1().Nodes().Get(assumedPod.Spec.NodeName, metav1.GetOptions{})
+	if err != nil {
+		return err
+	}
+
+	// Check node affinity on bound volumes
+	if ok, err := b.checkBoundClaims(claimsToProvision, node, assumedPod.GetName()); !ok {
+		errMsg := "failed validating PV node affinity"
+		if err != nil {
+			return fmt.Errorf("%s: %v", errMsg, err)
+		}
+		return fmt.Errorf(errMsg)
```

Comment:
would prefer this error message to be a little more detailed, like "PV %v node affinity does not match node %v"

Reply:
From here, I don't have access to the PV that failed the validation. Seems like I'd have to make `checkBoundClaims` return this info (for now it only logs it).

I updated the message to include the node name, but let me know if you think this isn't enough.

Reply:
If I understand correctly, there could be more than one PVC in `claims` (which is a list of PVCs) that can fail the node affinity check. From here I only know that something failed, it could be only one or even all of them.

`checkBoundClaims` doesn't return which ones failed, it only logs them. I'd have to change this method (and everything that's using it). Perhaps is that what you are suggesting?

Reply:
I think it is fine to checkBoundClaims to also return the first failed PV.  There's only one other caller.

Reply:
Perfect! Done.